vega_sim/scenario/common/agents.py

```diff
@@ -3175,3 +3177,209 @@ def initialise(
+        mint_key: bool = True,
+    ):
+        # Initialise wallet for LP/ Settle Party
+        super().initialise(vega=vega, create_key=create_key)
+
+        # Get asset id
+        self.asset_id = self.vega.find_asset_id(symbol=self.asset_name)
+        if mint_key:
+            # Top up asset
+            self.vega.mint(
+                wallet_name=self.wallet_name,
+                asset=self.asset_id,
+                amount=self.initial_asset_mint,
+                key_name=self.key_name,
+            )
+            self.vega.wait_for_total_catchup()
+
+        # Get market id
+        self.market_id = self.vega.find_market_id(name=self.market_name)
+
+        self._update_state(current_step=self.current_step)
+
+    def step(self, vega_state: VegaState):
+        self.current_step += 1
+
+        # Each step, get position
+        position = self.vega.positions_by_market(
+            wallet_name=self.wallet_name,
+            market_id=self.market_id,
+            key_name=self.key_name,
+        )
+        self.current_position = float(
+            position.open_volume if position is not None else 0
+        )
+
+        # Get a list of live orders for the party
+        orders = list(
+            (
+                vega_state.market_state[self.market_id]
+                .orders.get(
+                    self.vega.wallet.public_key(
+                        wallet_name=self.wallet_name, name=self.key_name
+                    ),
+                    {},
+                )
+                .values()
+            )
+            if self.market_id in vega_state.market_state
+            else []
+        )
+
+        # Check buy_order_reference and sell_order_reference are still live:
+        buy_seen = False
+        sell_seen = False
+        for order in orders:
+            if order.id == self.buy_order_reference:
+                buy_seen = True
+            if order.id == self.sell_order_reference:
+                sell_seen = True
+        if not buy_seen:
+            self.buy_order_reference = None
+        if not sell_seen:
+            self.sell_order_reference = None
+
+        # If there is position within limits we place both buy and sell
+        if (
+            -self.max_position <= self.current_position
+            and self.current_position <= self.max_position
+        ):
+            if self.buy_order_reference is None:
+                self.buy_order_reference = self.vega.submit_order(
+                    trading_wallet=self.wallet_name,
+                    trading_key=self.key_name,
+                    market_id=self.market_id,
+                    order_type="TYPE_LIMIT",
+                    time_in_force="TIME_IN_FORCE_GTC",
+                    side="SIDE_BUY",
+                    volume=self.order_size,
+                    price=None,
+                    expires_at=None,
+                    pegged_order=PeggedOrder(
+                        reference="PEGGED_REFERENCE_MID", offset=self.peg_offset
```

Comment:
These should be pegged to best bid/ask no?

Reply:
Not sure I could see an argument for both being useful, all okay if I make it configurable @TomMcL ?

Reply:
Configurable works too, at the touch generally means best bid/ask though so that should be default :)